extras/AFC.py

```diff
@@ -665,19 +666,17 @@ def cmd_CHANGE_TOOL(self, gcmd):
                     return
             CUR_LANE = self.printer.lookup_object('AFC_stepper ' + lane)
             self.TOOL_LOAD(CUR_LANE)
-            newpos = self.toolhead.get_position()
-            newpos[2] = store_pos[2]
-            self.toolhead.manual_move(newpos, self.tool_unload_speed)
-            self.toolhead.wait_moves()
-            if self.is_printing() and not self.is_paused():
-                self.change_tool_pos = None
+            # Restore state
+            self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
+            if self.is_printing() and self.is_paused():
+                self.failed_in_toolchange = True
 
     cmd_RESTORE_CHANGE_TOOL_POS_help = "change filaments in tool head"
     def cmd_RESTORE_CHANGE_TOOL_POS(self, gcmd):
-        if self.change_tool_pos:
-            restore_pos = self.change_tool_pos[:3]
-            self.toolhead.manual_move(restore_pos, self.tool_start_unload_speed)
-            self.toolhead.wait_moves()
+        if self.failed_in_toolchange:
+            # Restore previous state
+            self.failed_in_toolchange = False
+            self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
```

Comment:
While rebasing I noticed somewhere between my original PR and now this restore position macro started using a different parameter for the speed (`tool_start_unload_speed` vs `tool_unload_speed`) BUT I don't see `tool_start_unload_speed` actually defined anywhere, so I've restored it back to `self.tool_unload_speed`. But again as mentioned in another comment thread, we really probably want these restore position speeds to have their own configurable speed because the default `tool_unload_speed` may be too slow and lead to excess oozing during the restore.